Structure_Engine/Create/Physical/Wall.cs

```diff
@@ -0,0 +1,73 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+using BHPE = BH.oM.Physical.Elements;
+using BHPC = BH.oM.Physical.Constructions;
+using BH.oM.Physical.Reinforcement;
+using BH.oM.Structure.SurfaceProperties;
+using BH.oM.Structure.Reinforcement;
+using BH.oM.Structure.Fragments;
+using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Creates a physical Wall from a Panel. The Wall will be assigned a Construction based on the SurfaceProperty of the Panel.")]
+        [Input("panel", "The Panel to use as the base for the framing element.")]
+        [Output("wall", "The created physical Wall based on the Panel element provided.")]
+        public static BHPE.Wall Wall(Panel panel)
+        {
+            if (panel.IsNull())
+                return null;
+            //Get Construction
+            ISurfaceProperty prop = panel.Property;
+            BHPC.Construction construction = null;
+            if (prop == null)
+                Base.Compute.RecordWarning("The panel does not contain a surfaceProperty. Can not extract profile or material");
+            else
+                construction = panel.Property.IConstruction();
+            //Get Location
+            PolyCurve externalEdges = Geometry.Create.PolyCurve(panel.ExternalEdges.Select(x => x.Curve));
+            List<PolyCurve> internalEdges = panel.Openings.Select(opening => Geometry.Create.PolyCurve(opening.Edges.Select(edge => edge.Curve))).ToList();
+            BHPE.Wall surfaceElement = Physical.Create.Wall(construction, externalEdges, internalEdges);
+            string name = panel.Name ?? "";
+            surfaceElement.Name = name;
+            if (panel.FindFragment<PanelRebarIntent>() != null || panel.Property.FindFragment<ReinforcementDensity>() != null)
+            {
+                Base.Compute.RecordWarning("The panel has reinforcement, but this is not yet implemented.");
```

Comment:
Same as above, would specify the warning a bit more

Reply:
Agreed